db/change_cache.go

```diff
@@ -309,7 +309,7 @@ func (c *changeCache) waitForSequenceWithMissing(sequence uint64) {
 
 // Given a newly changed document (received from the tap feed), adds change entries to channels.
 // The JSON must be the raw document from the bucket, with the metadata and all.
-func (c *changeCache) DocChanged(docID string, docJSON []byte, seq uint64, vbNo uint16) {
+func (c *changeCache) DocChanged(docID string, docJSON []byte, seq uint64, vbNo uint16, dataType uint8) {
```

Comment:
Here's a more fleshed out example of what I mean: 

https://play.golang.org/p/MHOJcmx_-E

It works, but you are right in the sense that it doesn't fully enforce valid constant values.  

The value-add is that if someone is looking at:

```
func MemcachedDataTypeReceiver(fieldName string, dataType MemcachedDataType) {
	fmt.Printf("%s: %+v\n", fieldName, dataType)
}
```

it connects the parameter back to the `MemcachedDataType` constant and so anyone looking at that can easily see the named constant values (`MemcachedDataTypeJSON`, etc).  (IDE's will also allow jumping to the code or hovering over with a preview)